### PR TITLE
DX: check file permissions

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

**Comment by SpacePossum** [comment]
nice!
can you throw in to check `php-cs-fixer` in the root has `x` permission as well?

**Comment by SpacePossum** [comment]
I think adding another check (`if`) for the execution flag on that one file is fine, there is not much gain trying to merge it in the current logic think.

**Comment by SpacePossum** [comment]
please review my changes @kubawerlos

File: dev-tools/check_file_permissions.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+if [ "$files_with_wrong_permissions" ]
+then
+    printf '\033[97;41mWrong permissions detected:\033[0m\n'
+    e=$(printf '\033')
+    echo "${files_with_wrong_permissions}"
+    exit 3
+fi
+
+if [ -x "php-cs-fixer" ]
```
Comment (kubawerlos) [inline dev-tools/check_file_permissions.sh:20]: Oh, like this :) - I was try something `git ls-files` but always ending with silly/stupid command.

**Comment by kubawerlos** [comment]
Hold on, we need update `CONTRIBUTING.md` file.

**Comment by SpacePossum** [comment]
seems the `checkbashisms` script is broken itself;
https://travis-ci.org/github/FriendsOfPHP/PHP-CS-Fixer/jobs/734276849#L704

maybe fix;
```diff
--- dev-tools/bin/checkbashisms	2020-10-09 14:45:37.499597123 +0200
+++ dev-tools/bin/checkbashisms	2020-10-09 14:45:37.499597123 +0200
@@ -440,7 +440,7 @@
 	    # Finally the whole subexpression may be omitted for scripts
 	    # which do not pass on their parameters (i.e. after re-execing
 	    # they take their parameters (and potentially data) from stdin
-	    .?(\${1:?\+.?)?(\$(\@|\*))?~x) {
+	    .?(\$\{1:?\+.?)?(\$(\@|\*))?~x) {
             $ret = $. - 1;
             last;
         } elsif (/^\s*(\w+)=\$0;/) {
@@ -453,7 +453,7 @@
 	    # As above
 	    .?\$$var.?\s*
 	    (--\s*)?
-	    .?(\${1:?\+.?)?(\$(\@|\*))?.?\s*\&~x) {
+	    .?(\$\{1:?\+.?)?(\$(\@|\*))?.?\s*\&~x) {
 
 	    $backgrounded = 1;
 	} elsif ($backgrounded and m~


```

however they are not on github and my perl/regex/shell knowledge may not be enough, so I'll skip it for now

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Comment by keradus** [comment]
friendly note @SpacePossum , the `dev-tools/bin/checkbashisms` is file provided externally, it's not part of our repo

**Comment by SpacePossum** [comment]
Yeah I know, that is why I didn't change it in the repo. Still looks like a minor issue in script, but as they are not on github I don't know where to send the patch to

**Comment by keradus** [comment]
done, @SpacePossum : https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5215 ;)

**Comment by SpacePossum** [comment]
nice! thanks man : )

**Pull request merged**